adam-core/src/main/scala/org/bdgenomics/adam/rdd/ShuffleRegionJoin.scala

```diff
@@ -116,9 +120,11 @@ sealed trait ShuffleRegionJoin[T, U, RT, RU] extends RegionJoin[T, U, RT, RU] {
     if (leftIter.isEmpty || rightIter.isEmpty) {
       emptyFn(leftIter, rightIter)
     } else {
-      val bufferedLeft = leftIter.buffered
-      val currentBin = bufferedLeft.head._1._2
-      val region = bins.value.invert(currentBin)
+      val listRepresentationLeft = leftIter.toList
```

Comment:
This change is related to the problem of getting the ReferenceRegion that represents the partition. There is a better way to do this, so I will make some changes to the approach.